rclpy/rclpy/action/client.py

```diff
@@ -180,6 +181,7 @@ def __init__(
 
         callback_group.add_entity(self)
         self._node.add_waitable(self)
+        self._logger = get_logger("rclpy_action")
```

Comment:
```suggestion
        self._logger = self.node.get_logger().get_child('action_client')
```

Reply:
I think it makes sense to create a logger in the same namespace than the node logger, but not an independent one

Reply:
I see, but that makes my use case more difficult because now each logger has a different name. I need them to have the same name to filter them out more easily.

I'm able to filter out unwanted DEBUG logs like this:

```
log_options = [
    "--ros-args",
    "--log-level",
   "debug",
    "--log-level",
    "rmw_cyclonedds_cpp:=info",
    "--log-level",
    "rcl_action:=info",
    "--log-level",
    "rclpy_action:=info",
],

Node(
    package="some_package",
    executable="some_executable",
    arguments=log_options
),
```

With the change you propose, I'd have do replace `rclpy_action` with `namespace.some_executable.action_client`. Then each Node has a different argument

Reply:
In case you are wondering why I'm setting all the loggers to debug and then selectively setting some to info, instead of:

```
Node(
    package="some_package",
    executable="some_executable",
    arguments=[
    "--ros-args",
    "--log-level",
   "namespace.some_executable:=debug",
    ]
),
```

That's because:

1- To avoid setting different argument for each node
2- The namespace is a variable LaunchConfiguration (I guess parsing it is still possible)
3- I don't necessarily know all the name of all the loggers in the node

Reply:
Lastly, your proposition diverges from the rclcpp implementation [where an independent logger is used](https://github.com/ros2/rclcpp/blob/b953bdddf8de213b4a051ecf2d668dad65ff9f89/rclcpp_action/src/server.cpp#L100)

Reply:
> I see, but that makes my use case more difficult because now each logger has a different name. I need them to have the same name to filter them out more easily.

What if I only want to view the logs of one action server?

Which log messages are you hoping to see in your use case?

All entities seem to use the node's logger at at the moment ([I was surprised by this one](https://github.com/ros2/rclpy/blob/7685ee21a21fcb682ec42785544e03d05f83e851/rclpy/src/rclpy/subscription.cpp#L152-L159)). I'd lean towards nesting them further so that they become individually fiterable.

Reply:
> Lastly, your proposition diverges from the rclcpp implementation [where an independent logger is used](https://github.com/ros2/rclcpp/blob/b953bdddf8de213b4a051ecf2d668dad65ff9f89/rclcpp_action/src/server.cpp#L100)

That's an error message in a unique pointer deleter that doesn't have access to the logger. [rclcpp uses the node's logger name plus `rclcpp_action`](https://github.com/ros2/rclcpp/blob/b953bdddf8de213b4a051ecf2d668dad65ff9f89/rclcpp_action/src/server.cpp#L91) for [most log messages](https://github.com/ros2/rclcpp/blob/b953bdddf8de213b4a051ecf2d668dad65ff9f89/rclcpp_action/src/server.cpp#L355).

Reply:
Okay I understand better now. Having it as a child logger makes generally more sense. I'll find another workaround for my use case (or write another issue to support glob for selecting loggers)